pkg/reconciler/namespace/namespace.go

```diff
@@ -100,55 +103,156 @@ func (c *Controller) reconcileGVR(ctx context.Context, gvr schema.GroupVersionRe
 	return nil
 }
 
-// reconcileNamespace is responsible for assigning a namespace to a cluster, if
-// it does not already have one.
-//
-// After assigning (or if it's already assigned), this also updates all
-// resources in the namespace to be assigned to the namespace's cluster.
-func (c *Controller) reconcileNamespace(ctx context.Context, lclusterName string, ns *corev1.Namespace) error {
-	klog.Infof("Reconciling Namespace %s", ns.Name)
+func (c *Controller) patchStatus(ctx context.Context, ns *corev1.Namespace, updateConditions updateConditionsFunc) error {
+	patchBytes, err := statusPatchBytes(ns, updateConditions)
+	if err != nil {
+		return err
+	}
+	_, err = c.namespaceClient.Patch(ctx, ns.Name, types.MergePatchType, patchBytes, metav1.PatchOptions{}, "status")
+	if err != nil {
+		return fmt.Errorf("failed to patch status on namespace %q: %w", ns.Name, err)
+	}
+	return nil
+}
 
-	if ns.Labels == nil {
-		ns.Labels = map[string]string{}
+// Set the schedulable condition on namespaces and not the resources it
+// contains. Not all native types support status conditions (e.g. configmaps
+// and secrets) and there is no guarantee that arbitrary CRDs will support
+// conditions.
+func (c *Controller) updateSchedulableCondition(ctx context.Context, ns *corev1.Namespace, schedulable bool) error {
+	klog.Infof("Patching namespace %q status to indicate schedulable=%s", ns.Name, schedulable)
```

Comment:
do we know here that there is a diff? Otherwise, this is noisy.

Reply:
As per `ensureScheduledStatus`, `updateScheduledCondition` is only called if a) there was a change in scheduled cluster or b) the current condition doesn't match the state of the namespace label.  I'm not sure I see your concern?

Reply:
The check now ensures status is only updated if the namespace goes from scheduled -> not scheduled or vise versa.